bokeh/models/ranges.py

```diff
@@ -42,6 +42,16 @@ class Range1d(Range):
     The end of the range.
     """)
 
+    bound_lower = Either(Float, Datetime, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
The `Datetime` in here really bugs me. I'm not sure what a better solution would be offhand.